in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,239 @@
+  rsa_key = in_toto.util.prompt_import_rsa_key_from_file(key)
+
+  # Checking if the given key follows the format
+  securesystemslib.formats.KEY_SCHEMA.check_match(rsa_key)
+
+  signable_object.sign(rsa_key)
+
+  return signable_object
+
+
+def replace_sign(link, key):
+  """
+  <Purpose>
+    Replaces all the existing signature with the new signature,
+    signs the file, and then returns the link file as an object.
+
+  <Arguments>
+    link - path to the key file
+    key - the key to be used for signing
+
+  <Exceptions>
+    None
+
+  <Returns>
+    An object containing the contents of the link file after
+    adding the signature which replaces the old signatures
+  """
+
+  # Reading the link file and making an object
+  signable_object = link_import.read_from_file(link)
+
+  # Import rsa key from the filepath
+  rsa_key = in_toto.util.prompt_import_rsa_key_from_file(key)
+
+  # Check if the key corresponds to the correct format
+  securesystemslib.formats.KEY_SCHEMA.check_match(rsa_key)
+
+  # Remove all the existing signatures
+  signable_object.signatures = []
+
+  # Sign the object
+  signable_object.sign(rsa_key)
+
+  return signable_object
+
+
+def verify_sign(link, key_pub):
+  """
+  <Purpose>
+    Verifies the signature field in the link file, given a public key
+
+  <Arguments>
+    link - path to the link file
+    key_pub - public key to be used to verification
+
+  <Exceptions>
+    Raises SignatureVerificationError
+      - 'Invalid Signature' : when the verification fails
+      - 'Signature Key Not Found' : when KeyError occurs
+      - 'No Signatures Found' - when no signature exists
+
+  <Returns>
+    None
+  """
+  signable_object = link_import.read_from_file(link)
+  link_key_dict = in_toto.util.import_rsa_public_keys_from_files_as_dict(
+    [key_pub])
+  signable_object.verify_signatures(link_key_dict)
+
+def parse_args():
+  """
+  <Purpose>
+    A function which parses the user supplied arguments.
+
+  <Arguments>
+    None
+
+  <Exceptions>
+    None
+
+  <Returns>
+    Parsed arguments (args object)
+  """
+  parser = argparse.ArgumentParser(
+    description="in-toto-sign : Signs link file with/without replacement "
+                "of the existing signatures and dumps with/without "
+                "infixing the keyID in the filename")
+
+  lpad = (len(parser.prog) + 1) * " "
+
+  parser.usage = ("\n"
+                  "(sign | verify) --key <filepath>\n{0}"
+                  "[--replace-sig]\n{0}"
+                  "[--infix]\n{0}"
+                  "[--verbose]\n{0}"
+                  "<path/to/signable>\n\n"
+                  .format(lpad))
+
+  in_toto_args = parser.add_argument_group("in-toto-sign options")
+
+  in_toto_args.add_argument("operator", type=str, choices=['sign', 'verify'],
+                            help="Sign or Verify")
+
+  in_toto_args.add_argument("-k", "--key", type=str, required=True,
+                            help="Path to the key ")
```

Comment:
Sure. But, while verifying, in case of multiple signatures, the public keys need to be input in the same order as it was signed by the functionaries. That is, for example, the file was first signed by A, then by B (without replacement), then by C (without replacement). When verification takes place, then, pub key of A, followed by B, followed by C, should be specified in the arguments. Right? Or? 

Also there will be multiple cases, depending upon if any one of the functionaries in the supply chain have replaced all the existing signatures or not. What do you say?